include/zephyr/posix/sys/select.h

```diff
@@ -6,7 +6,7 @@
 #ifndef ZEPHYR_INCLUDE_POSIX_SYS_SELECT_H_
 #define ZEPHYR_INCLUDE_POSIX_SYS_SELECT_H_
 
-#include "posix_types.h"
+#include <zephyr/posix/posix_types.h>
```

Comment:
@keith-packard does have a point, and this line is the only one that needs changing, since it's in a different directly than `posix_types.h`. 

That being said, I do think we should use the full include path for these includes, as that is more future proof in case any of these files are ever moved (it is also the most common approach in Zephyr, and how tools like `clangd` Include Cleaner (and others) would include them.

Reply:
Yeah. I was aware of this but felt for consistency we should just change all of them.